modules/tinymce/src/plugins/fullscreen/main/ts/api/Events.ts

```diff
@@ -1,10 +1,25 @@
+import { Cell } from '@ephox/katamari';
+
 import Editor from 'tinymce/core/api/Editor';
+import Delay from 'tinymce/core/api/util/Delay';
+
+import * as Options from '../api/Options';
+import * as Actions from '../core/Actions';
 
 const fireFullscreenStateChanged = (editor: Editor, state: boolean): void => {
   editor.dispatch('FullscreenStateChanged', { state });
   editor.dispatch('ResizeEditor');
 };
 
+const setup = (editor: Editor, fullscreenState: Cell<Actions.FullScreenInfo | null>): void => {
+  editor.on('blur', () => {
+    if (fullscreenState.get() && Options.shouldFullScreenTrapFocus(editor)) {
+      Delay.setEditorTimeout(editor, () => editor.focus());
```

Comment:
Will this flicker. Can this be done instead by preventing tab key default behavior in the editor if the editor is in a fullscreen state instead?

Reply:
@spocke I was thinking the same initially, but won't preventing tab key default behaviour would also interfere with the list indentation, nonbreaking plugin tab and table navigation? Not sure if I'm missing something.

Reply:
It would but I guess it depends on when you register that block if it's after all those then the defulat native behavior would be prevented. Would have been nice if we had a preventDefaultNative so that we would only prevent default behavior but not our behavior. 😊

Reply:
Yeah if you wait for the `beforeInit` or `init` events to register a key handler it will happen after the normal handlers (core and other plugins) are registered.

But `blur` only happens when tab causes focus to leave the editor, so there should be no need to delay this focus call?

Reply:
Fixed that now the logic resides in core, so there's no need of this event here in the plugin 👍